pylint/checkers/variables.py

```diff
@@ -1805,7 +1805,7 @@ def _is_first_level_self_reference(
         """
         if (
             node.frame().parent == defstmt
-            and node.statement(future=True) not in node.frame().body
+            and node.statement(future=True) == node.frame()
```

Comment:
I don't understand how this work, could you explain ?

Reply:
Basically for arguments in function definitions their statement is also their frame (the function definition). For nodes within the body or even within sub-bodies (conditionals) the statement will be different from the frame (the latter still being the function definition).

Reply:
That makes sense, thank you.